R/db-sql.R

```diff
@@ -796,3 +804,35 @@ dbplyr_sql_subquery <- function(con, ...) {
 sql_subquery.DBIConnection <- function(con, from, name = unique_subquery_name(), ..., lvl = 0) {
   sql_query_wrap(con, from = from, name = name, ..., lvl = lvl)
 }
+
+safe_db_get_query <- function(con, sql, ..., .msg, .call = caller_env()) {
+  glue_envir <- caller_env()
+  tryCatch(
+    DBI::dbGetQuery(con, sql, ...),
+    error = function(cnd) {
+      cli_abort(
+        .msg,
+        parent = cnd,
+        .envir = glue_envir,
+        call = .call,
+        class = "dbplyr_error_dbi"
```

Comment:
Not sure whether you want a classed error here or not. I added this for testing but we might want to keep it. What do you think?

Reply:
I think our experience so far is that classed errors tend to be more trouble than their worth, so I'd omit it for now.

I'm also not sure if these helper functions should exist — something about the way that `glue_envir` is needed makes me wonder if it would be better to just have individual `tryCatch()`s in each of the functions.